api/src/main/java/org/apache/iceberg/io/CloseableIterable.java

```diff
@@ -117,6 +118,12 @@ public O next() {
     };
   }
 
+  static <E> CloseableIterable<E> concat(Iterable<? extends Iterable<E>> iterable,
```

Comment:
This doesn't concatenate the iterables, it mixes them together, so we should come up with a better name. Maybe `combine`?

Reply:
Agree! I prefer `combine`, because user can set `executorService` to null to run task serially.